### ISSUE TITLE
Support for $ref to external file broken in 2.7.8

### ISSUE THREAD

**Issue by bburtin** [LucyBot-Inc/api-spec-converter#166]
The recent changes in api-spec-converter 2.7.8 broke my ability to reference an external JSON schema.  Reverting to 2.7.7 solved my problem.  Would be great to get this fixed in the next release.  Thanks in advance!
```
$ node_modules/.bin/api-spec-converter --from openapi_3 --to swagger_2 v2api.yaml
Fatal Error: Error during conversion: Could not parse $ref Id.yaml as JSON or YAML
...
```
Here's the relevant snippet from my OpenAPI schema file.  I tried both "Id.yaml" and "./Id.yaml".  They both fail in the same way.

```
paths:
  '/api/v2/user/{id}':
    get:
      tags:
        - User
      summary: Get user
      operationId: user.get
      parameters:
        - name: id
          description: >-
            The user ID, email address, or `me`
            for the currently authenticated user
          in: path
          required: true
          schema:
            $ref: Id.yaml
      responses:
        '200':
          description: ''
          content:
            application/json:
              schema:
                $ref: User.yaml
        '400':
          description: "`InvalidUserId`"
          content:
            application/json:
              schema:
                $ref: ApiError.yaml
        '404':
          description: ''
```

**Comment by bobby-brennan** [comment]
Can you share your external YAML files as well?

Are the external $refs actually getting included when you use 2.7.7, or are they being silently ignored?

**Comment by bburtin** [comment]
Here's a minimal schema that reproduces the problem.
`~/B/w/a/issue166 (api-spec-converter-issue-166|✔) $ cat issue166.yaml
openapi: 3.0.0

paths:
  '/api/v2/user/{id}':
    get:
      tags:
        - User
      summary: Get user
      operationId: user.get
      parameters:
        - name: id
          in: path
          schema:
            $ref: Id.yaml
~/B/w/a/issue166 (api-spec-converter-issue-166|✔) $ cat Id.yaml
type: string
~/B/w/a/issue166 (api-spec-converter-issue-166|✔) $
../node_modules/.bin/api-spec-converter --from openapi_3 --to swagger_2 issue166.yaml
Fatal Error: Error during conversion: Could not parse $ref Id.yaml as JSON or YAML
    at Converter.resolveReference (/Users/boris/BitGo/www/api-schema/node_modules/api-spec-converter/lib/converters/openapi3_to_swagger2.js:82:15)
...`

**Comment by fenollp** [comment]
How about `$ref: 'Id.yaml#'` instead of just  `$ref: Id.yaml`?

**Comment by bobby-brennan** [comment]
Thanks for the details, Boris. This should be fixed in 2.7.9 - let me know if you are still seeing issues.

**Comment by bburtin** [comment]
Thanks Bobby.  After running *npm install LucyBot-Inc/api-spec-converter#v2.7.9*, I still get a parsing failure.  I modified the code, so that it logs the exception (the XXX part below) and see this:
```
~/issue166 $ node_modules/.bin/api-spec-converter --from openapi_3 --to swagger_2 issue166.yaml
XXX parsing failed ReferenceError: YAML is not defined
    at Converter.resolveReference (/Users/boris/issue166/node_modules/api-spec-converter/lib/converters/openapi3_to_swagger2.js:81:20)
```
Looks like a config issue with YAML parsing?  If I reference a JSON file, it works fine.

**Comment by bobby-brennan** [comment]
My fault - I only tested JSON, not YAML. Pushed a fix and a YAML test case in 2.7.10 - hopefully that does the trick.

**Comment by bburtin** [comment]
That did it!  Thanks for the quick fix.
